UI: Add warning for recording to mp4 format

derrod opened the pull request:
While recording to mp4 in obs-studio is safer than in classic there still are situations where a file is not finalized such as

* Some crashes to desktop as well as situation where obs becomes unresponsive
* Crashes outside of obs (Bluescreens, Power outages, etc.)
* Recordings never stopping i.e. Button saying "Stopping Recording..." but it never finalizing (At least on once occurence forceful stopping as introduced in 5c5d0ba9e5055ea4fc961b06d03263ed10920a96 didn't work for me either)
* File system is full and file can no longer be written

Given the amount of users I've seen in the past weeks having issues with unfinalized recordings I think it is sensible to warn them about the possible dangers of recording to mp4 while also suggesting an alternative (mkv + remuxing).

The warning message and the way I'm adding the warning could be improved. I'm not sure if splitting this up into `advOutRecWarning` and `advOutRecError` is the best option here.

ONSBalder commented:
Can one of the admins verify this patch?

SuslikV commented:
Should be closed by the same reason: 

> no change required

 https://obsproject.com/mantis/view.php?id=724&nbn=2

derrod commented:
Changed the code a bit to no longer require an additional `QLabel` since we're only using one anyways (error has priority over warning). Also removed the second if clause and used ternary operators instead.

Also changed the wording of the warning message a bit.
